model-optimizer/mo/back/preprocessing.py

```diff
@@ -0,0 +1,381 @@
+    layout_str = layout_str[:dim_idx_found] + 'C' + layout_str[dim_idx_found+1:]
+    layout_values[name] = {
+        'source_layout': layout_str,
+        'target_layout': None,
+        'source_guessed': True,
+        'is_input': True
+    }
+    return layout_values
+
+
+def guess_source_layouts_by_mean_scale(ov_function: Function, layout_values, mean_scale_values: dict):
+    """
+    Internal function. Try to guess source layout for input by its shape and/or framework
+    :param: ov_function Original model
+    :param: layout_values Existing source/target layout items specified by user
+    :param: mean_scale_values Dictionary with mean/scale values defined for each argument
+    :return: updated layout items with guessed layouts
+    """
+    for ms_name, mean_scale in mean_scale_values.items():
+        num_channels_mean = len(mean_scale['mean']) if mean_scale['mean'] is not None else 0
+        num_channels_scale = len(mean_scale['scale']) if hasattr(mean_scale['scale'], '__len__') else 0
+
+        if num_channels_mean > 1 and \
+                num_channels_scale > 1 and \
+                num_channels_mean is not num_channels_scale:
+            raise Error('Mean/Scale values for {} have different sizes: {} {}'
+                        .format(ms_name, num_channels_mean, num_channels_scale))
+
+        need_guess_channels = num_channels_mean > 1 or num_channels_scale > 1
+        if not need_guess_channels:  # Mean/scale is complex and needs 'channels' specified in layout
+            continue
+
+        num_channels = num_channels_mean if num_channels_mean > 1 else num_channels_scale
+
+        for i in range(0, len(ov_function.inputs)):
+            ov_input = ov_function.input(i)
+
+            if not ov_function.get_parameters()[i].layout.empty:
+                continue
+
+            if ms_name not in ov_input.get_tensor().get_names():
+                continue
+
+            layout_item = None
+            for name in ov_input.get_tensor().get_names():
+                if name in layout_values:
+                    layout_item = layout_values[name]
+                    break
+
+            if layout_item is not None:
+                # User specified some layout, skip guessing
+                continue
+
+            if num_channels != 3:
+                raise Error('Can\'t determine channels dimension for {}. '
+                            'When number of mean/scale values is {} (not 3), '
+                            'please specify layout for input manually'.format(ms_name, num_channels))
+
+            layout_values = find_channels_dimension(shape=ov_input.get_partial_shape(),
+                                                    num_channels=num_channels,
+                                                    name=ms_name,
+                                                    layout_values=layout_values)
+    return layout_values
+
+
+def check_suitable_for_reverse(layout: Layout, ov_input):
```

Comment:
please add type for `ov_input`

Reply:
`ov::Output<ov::Node>` is not publicly exposed in python currently. It is declared in openvino.runtime.impl but is not exposed further. This will require to add 'from openvino.runtime.impl import Output' and code will have `ov_input: Output` which looks quite strange for those who are not familiar with 'ngraph' structure